### DISCUSSION TITLE
Create repositories for autogenerated clients from openapi

### DISCUSSION THREAD

**Discussion by hoshsadiq** [firefly-iii/firefly-iii#5027] [Firefly III] [answered]
I'm getting slowly more accustomed to Firefly III and currently I'm trying to write a custom importer. Specifically one that automatically adds transactions from email receipts. This will hopefully be extensible for additional providers. The aim being that, when someone sets this up correctly, transactions are automatically added to Firefly III as receipts are received in your inbox. This is ideal for fintech that allow for sending receipts to your email for all transactions (e.g. PayPal, Curve Bank) as well as retailers (most of whom send receipts).

Anyway, as part of this, I'm trying to invoke the Firefly III API, and while the docs are great, it is still a pain to have to write the API requests manually (even if abstracted). I noticed you publish [OpenAPI spec](https://api-docs.firefly-iii.org/) which can be used to generate an SDK for different languages. It would be great for some common SDKs to be auto-generated under the Firefly-III org. In general it _shouldn't_ need much maintenance as it should be automatically be generated. It can even be auto invoked if the api-docs repo is tagged or something (I'm not sure where the pipeline for it is).

For the languages _I_ need, I've seen several already, but I'd much rather use a library in direct control of the original maintainers of a project (because trust). If you're not okay with this, I'll probably end up doing this for my needed libraries in a repo for myself. If you are, I'd be happy to set the repo up and transfer ownership, and subsequently submit automation PRs. From there it is up to you if you'd like me to continue to maintain only those repos or just let others contribute.

Let me know what you think.

**Comment by JC5** [comment, marked as answer]
I see what you need, but for the time being, I'm not going to offer auto-generated clients. I don't want to have the maintenance burden, especially not for languages I'm not familiar with. Rather, I'd focus on the source itself (ie. the API docs) and maintain those as best I can. That's proving to be hard enough already 😉

**Reply by hoshsadiq** [reply]
No worries. That's fair. I'd be happy to maintain them a few client SDKs if that's the only issue. If not, no worries either way.
